task/build-maven-zip/0.1/build-maven-zip.yaml

```diff
@@ -0,0 +1,210 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "maven-zip"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "maven-build, konflux"
+  name: build-maven-zip
+spec:
+  description: |-
+    This takes existing Image Manifests and combines them in an Image Index.
+  params:
+  - description: Reference of the OCI-Artifact this build task will produce.
+    name: IMAGE
+    type: string
+  - default: "generic"
+    description: The prefetched content which is used in the build.
+    name: PREFETCH_INPUT
+    type: string
+  - default: "maven-repository"
+    description: The root of the prefetched artifacts
+    name: PREFETCH_ROOT
+    type: string
+  - default: "maven-repository"
+    description: The zip bundle name of archived artifacts
+    name: BUNDLE_NAME
+    type: string
+  - default: ""
+    description: Delete image tag after specified time. Empty means to keep the image tag. Time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
+    name: IMAGE_EXPIRES_AFTER
+    type: string
+  - name: caTrustConfigMapName
+    type: string
+    description: The name of the ConfigMap to read CA bundle data from.
+    default: trusted-ca
+  - name: caTrustConfigMapKey
+    type: string
+    description: The name of the key in the ConfigMap that contains the CA bundle data.
+    default: ca-bundle.crt
+
+  results:
+  - description: Digest of the image just built
+    name: IMAGE_DIGEST
+  - description: Image repository and tag where the built image was pushed
+    name: IMAGE_URL
+  - name: SBOM_BLOB_URL
+    description: Reference of SBOM blob digest to enable digest-based verification from provenance
+    type: string
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /shared
+        name: shared
+    env:
+    - name: IMAGE
+      value: $(params.IMAGE)
+    - name: INPUT
+      value: $(params.PREFETCH_INPUT)
+    - name: PKG_ROOT
+      value: $(params.PREFETCH_ROOT)
+    - name: BUNDLE_NAME
+      value: $(params.BUNDLE_NAME)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.IMAGE_EXPIRES_AFTER)
+
+  steps:
+  - image: quay.io/konflux-ci/appstudio-utils@sha256:d0fecb67dd98d874704f2e3d20686363de7cbb42c15ad2b1d8e8c38118c50078
+    name: prepare
+    computeResources:
+      limits:
+        memory: 8Gi
+        cpu: '4'
+      requests:
+        memory: 2Gi
+        cpu: '1'
+    script: |
+      #!/bin/bash
+      set -o pipefail
+      # Generate checksums for all maven artifact files. It will ignore the checksum files
+      # and signature files if they existed there
+      pkgpath="$(workspaces.source.path)/cachi2/output/deps/$INPUT"
```

Comment:
possible forms of `INPUT` at the moment:
1. just a package manager name, e.g. `generic`
2. `{"type": "<package manager name>", <more attributes>}`
3. an array of the objects from above, e.g. `[{"type": "generic", ...}, {"type": "rpm", ...}, ...]`
4. `{"packages": [<an array like above>], "flags": [<supported flags>]}`

This only supports the simplest one

Reply:
TBH, this task probably shouldn't take `PREFETCH_INPUT` into account at all, it should only handle `cachi2/output/deps/generic`

Reply:
Sounds ok. I'll use it directly.

Reply:
Fixed in new commit